src/strategies/strategy-rally-farm-base.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.7;
+
+import "./strategy-base.sol";
+import "../interfaces/rally-chef.sol";
+
+abstract contract StrategyRallyFarmBase is StrategyBase {
+    // Token addresses
+    address public constant rally = 0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b;
+    address public constant rallyChef = 0x9CF178df8DDb65B9ea7d4C2f5d1610eB82927230;
+
+    // WETH/<token1> pair
+    address public token1;
+
+    uint256 public poolId;
+
+    constructor(
+        address _token1,
+        uint256 _poolId,
+        address _lp,
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    )
+        public
+        StrategyBase(_lp, _governance, _strategist, _controller, _timelock)
+    {
+        poolId = _poolId;
+        token1 = _token1;
+    }
+
+    function balanceOfPool() public override view returns (uint256) {
+        (uint256 amount, ) = IRallychef(rallyChef).userInfo(
+            poolId,
+            address(this)
+        );
+        return amount;
+    }
+
+    function getHarvestable() external view returns (uint256) {
+        return IRallychef(rallyChef).pendingRally(poolId, address(this));
+    }
+
+    // **** Setters ****
+
+    function deposit() public override {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            IERC20(want).safeApprove(rallyChef, 0);
+            IERC20(want).safeApprove(rallyChef, _want);
+            IRallychef(rallyChef).deposit(poolId, _want);
+        }
+    }
+
+    function _withdrawSome(uint256 _amount)
+        internal
+        override
+        returns (uint256)
+    {
+        IRallychef(rallyChef).withdraw(poolId, _amount);
+        return _amount;
+    }
+
+    // **** State Mutations ****
+
+    function harvest() public override onlyBenevolent {
+        // Collects RLY tokens
+        IRallychef(rallyChef).deposit(poolId, 0);
+        uint256 _rally = IERC20(rally).balanceOf(address(this));
+
+        // Swap half to WETH
+        if (_rally > 0) {
+            IERC20(rally).safeApprove(univ2Router2, 0);
+            IERC20(rally).safeApprove(univ2Router2, _rally.div(2));
+            _swapUniswap(rally, weth, _rally.div(2));
+        }
+
+        if (token1 != rally) {
```

Comment:
@larrythecucumber321 can `token1` be anything else than RALLY? And if not, should we call it `rally` (and don't handle the case when it isn't RALLY)?

Reply:
@janklimo you're probably right that `token1` can probably just be `rally`. However, the way we've been doing things is to have a "base-strategy" for a particular protocol, and to implement the specific token/pools we farm on that protocol (for the purposes of maintaining modularity). See for example FOX-ETH:
https://github.com/pickle-finance/protocol/blob/master/src/strategies/strategy-fox-farm-base.sol
https://github.com/pickle-finance/protocol/blob/master/src/strategies/fox/strategy-fox-eth-fox-lp.sol

This structure makes a lot more sense for something like Sushi where we'll farm multiple ones of their pools. But for something where we're probably anticipating farming only one of their pools (like FOX or RLY), a monolithic strategy file (which ditches a base) might make more sense. We can do this going forward :)